crates/c-api/src/config.rs

```diff
@@ -255,3 +259,139 @@ pub unsafe extern "C" fn wasmtime_config_cranelift_flag_set(
     let value = CStr::from_ptr(value).to_str().expect("not valid utf-8");
     c.config.cranelift_flag_set(flag, value);
 }
+
+pub type wasmtime_new_memory_callback_t = extern "C" fn(
+    ty: wasm_memorytype_t,
+    minimum: usize,
+    maximum: usize,
+    reserved_size_in_bytes: usize,
+    guard_size_in_bytes: usize,
+    memory_ret: &mut *mut std::ffi::c_void,
+    finalizer_ret: &mut Option<extern "C" fn(arg1: *mut std::ffi::c_void)>,
+) -> Option<Box<wasmtime_error_t>>;
+
+pub type wasmtime_memory_get_callback_t = extern "C" fn(
+    memory_ptr: *mut std::ffi::c_void,
+    byte_size: &mut usize,
+    maximum_byte_size: &mut usize,
+) -> *mut u8;
```

Comment:
The return type in the C header is `void*` but here we have ` *mut u8`. For consistency, either

* this should change to `*mut std::ffi::c_void`
* or the C header should change to `uint8_t*`

Reply:
I changed the C header. Nice catch!